__tests__/app.test.js

```diff
@@ -149,6 +149,24 @@ describe("GET /api/articles", () => {
         expect(body.articles).toBeSortedBy("votes", { ascending: true });
       });
   });
+  test("200: filters articles by topic when topic query is provided", () => {
+    return request(app)
+      .get("/api/articles?topic=cats")
+      .expect(200)
+      .then(({ body }) => {
+        body.articles.forEach((article) => {
+          expect(article.topic).toBe("cats");
+        });
+      });
+  });
+  test("404: responds with 'topic not found' for non-existent topic", () => {
+    return request(app)
+      .get("/api/articles?topic=nonexistent")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body).toEqual({ message: "topic not found" });
+      });
```

Comment:
Nice test for a topic that doesn't exist 🥇 

There is possibly 1 more condition we need to check, what happens if a topic does exist but it doesn't have any associated articles yet? Would we still expect an error?